week2/assignment2-2/src/calculations.cpp

```diff
@@ -0,0 +1,57 @@
+#include "../includes/calculations.h"
+
+void get_numbers(double& num1, double& num2) {
+    std::cout << "Enter 1st integer: ";
+    std::cin >> num1;
+    if(!std::cin) { // User didn't input number -> cin fails
+        std::cin.clear();  // clear cin's failbit
+        std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n'); //skip bad input
+        while (true) {  // Keep asking valid input
+            std::cout << "Invalid input! Please try again.\n";
+            std::cout << "Enter 1st integer: ";
+            std::cin >> num1;
+            if(!std::cin) {
+                std::cin.clear();
+                std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+            } else {
+                break;
+            }
+        }
+    }
+    assert((typeid(num1) == typeid(double)) && "Input has to be number!\n"); 
+    
+    std::cout << "Enter 2nd integer: ";
+    std::cin >> num2;
+    if(!std::cin) { // User didn't input number -> cin fails
+        std::cin.clear();  // clear cin's failbit
+        std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n'); //skip bad input
+        while (true) {  // Keep asking valid input
+            std::cout << "Invalid input! Please try again.\n";
+            std::cout << "Enter 2st integer: ";
+            std::cin >> num2;
+            if(!std::cin) {
+                std::cin.clear();
+                std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+            } else {
+                break;
+            }
+        }
+    }
+    assert((typeid(num2) == typeid(double)) && "Input has to be number!\n"); 
+}
+
+void addition(double& num1, double& num2) {
+    std::cout << "\nThe sum of the numbers is " << (num1 + num2) << "\n\n";
```

Comment:
Näistä jo kerettiinki jossain kohtaa keskustella, mutta kun tehdään näitä functioita varsinkin "kirjastossa" niin näissä ei pitäisi olla käyttöliittymään mitään osviittaa. Logituksia virheistä tai muusta on toki suotavaa. 

Nämä funktioit pitäisi olla muotoa 
```
double addition(double& num1, double% num2) {
    return num1+num2;
}
```
Ja sitte printata tulos siellä mistä tätä funktiota kutsutaan.